pandas/io/pytables.py

```diff
@@ -316,9 +328,15 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         store = path_or_buf
         auto_close = False
+
     else:
-        raise NotImplementedError('Support for generic buffers has not been '
-                                  'implemented.')
+        path_or_buf = io_common._stringify_path(path_or_buf)
+        if isinstance(path_or_buf, string_types):
+            store = HDFStore(path_or_buf, **kwargs)
+            auto_close = True
+        else:
+            raise NotImplementedError('Support for generic buffers has not '
```

Comment:
test_pytables.TestHDFStore.test_read_hdf_errors should be hitting the else-branch (NotImplementedError), as the path_or_buf object is a regular file pointer (io.TextWrapper) there.  

pytables.TestHDFStore.test_read_from_pathlib_path should be hitting the if-branch on python3 (and on python2 if pathlib is installed).  

I apologize for my tardiness in responding to these...